server/komodo-rest/src/main/java/org/komodo/rest/TeiidSwarmMetadataInstance.java

```diff
@@ -260,9 +262,9 @@ private void createDataSource(String name, SourceType type, Map<String, String>
 			 */
 			Properties props = new Properties();
 			props.setProperty("connection-url", "jdbc:mysql://"+map.get("DATABASE_SERVICE_NAME")+":3306/" 
-					+ map.get("database-name"));
-			props.setProperty("user-name", map.get("database-user"));
-			props.setProperty("password", map.get("database-password"));
+					+ map.get("database_name"));
+			props.setProperty("user-name", map.get("username"));
+			props.setProperty("password", map.get("password"));
```

Comment:
Are these property names not guaranteed to exist then, ie. openshift service dependent?

Reply:
I would say as soon as we integrated with Service Catalog we were OpenShift dependent, but IMO there is nothing wrong with that. What we are seeing is an effect of multiple different templates for MySQL. We need to think little more about how to handle it.

I would even go further and say that we should tailor more knowing the platform and remove any APIs/code that is no longer needed. I am not saying there is, but it was there..

Reply:
@rareddy you misunderstand "openshift service dependent". I mean in the context of different implementations of mysql instances or postgresql instances in Openshift not independent of Openshift. Your comment goes on to illustrate my question in more detail.

Certainly need to think a little more as there is no guarantee that any one template is going to conform to a particular set of property names. For all we know "user" could be "username" or "uname" or "uow" or even "namespace" or "blah" (probably not the latter). That's going to require some serious handling!